lib/crystalball/rspec/runner.rb

```diff
@@ -70,44 +72,21 @@ def config_file
 
         def build_prediction(out)
           check_map(out)
-          # Actual examples size is not less than prediction size.
           prune_prediction_to_limit(prediction_builder.prediction.sort_by(&:length), out)
         end
 
-        def prune_prediction_to_limit(prediction, out)
-          limit = config['examples_limit'].to_i
-
-          return prediction if !limit.positive? || prediction.size <= limit
-
-          out.puts "Prediction size #{prediction.size} is over the limit (#{limit})"
-          out.puts "Prediction is pruned to fit the limit!"
-
-          prediction.first(limit)
-        end
-
         def check_map(out)
           out.puts 'Maps are outdated!' if prediction_builder.expired_map?
         end
       end
 
-      def run(err, out)
-        setup(err, out)
-        run_specs(prediction_example_groups(out)).tap do
-          persist_example_statuses
-        end
-      end
-
-      def prediction_example_groups(out)
-        limit = self.class.config['examples_limit'].to_i
-
-        pruner = ExamplesPruner.new(@world, to: limit)
-
-        return pruner.world_groups if !limit.positive? || @world.example_count(pruner.world_groups) <= limit
+      def setup(err, out)
+        super
 
-        out.puts "Prediction examples size #{@world.example_count(pruner.world_groups)} is over the limit (#{limit})"
-        out.puts "Prediction is pruned to fit the limit!"
+        return unless reconfiguration_needed?
 
-        pruner.pruned_groups
+        reconfigure_to_limit
+        super
```

Comment:
Why are you calling `super` twice?

Reply:
Because RSpec needs to be reconfigured again after changing filters. I reworked a bit so there is no `super` calls anymore